internal/cluster/connection_listener_binder.go

```diff
@@ -225,7 +218,14 @@ func (b *ConnectionListenerBinder) sendRemoveListenerRequest(ctx context.Context
 		return fmt.Sprintf("%d: removing listener", conn.connectionID)
 	})
 	inv := b.invocationFactory.NewConnectionBoundInvocation(request, conn, nil, start)
-	if err := b.invocationService.SendRequest(ctx, inv); err != nil {
+	req := inv.Request()
+	_, err := b.connectionManager.invoker.CB().TryContext(ctx, func(ctx context.Context, attempt int) (interface{}, error) {
```

Comment:
Listener registration / deregistration is currently not urgent in the Go client because of that.

Reply:
We made the listener invocations urgent in Java because they are connection-bound requests. And, we would throw IOError in case of trying to send a non-urgent invocation while the client is not fully connected. We do not retry invocations that receive IOException, if they are connection bound in Java. Is this not the case in Go? What do we do differently here that allows us to get away with non-urgent listener registrations?

Reply:
`b.connectionManager.invoker.CB().TryContext` retries sending the invocation, if the invocation is rejected due to invocation service is paused.

Reply:
I see that, but normally, IOException should be terminal for the connection bound invocations. Is this not the case in the Go client?

Reply:
`SendRequest` just puts the invocation to a channel to be processed. If the invocation service is paused it directly returns an error. That's the only difference with `SendUrgentRequest`, which wouldn't return an error in this case.  `b.connectionManager.invoker.CB().TryContext` retries only the `SendRequest` bit.

IOException is supposed to be not retried in the Go client too.